sw/device/lib/testing/test_framework/freertos_port.S

```diff
@@ -15,6 +16,36 @@
 
 // -----------------------------------------------------------------------------
 
+/**
+ * Save MEPC to the stack.
+ *
+ * NOTE: this IRQ is synchronous, therefore, we must update the ISR return
+ * address to point to the instruction after the one that triggered this IRQ.
+ * Since we support the RISC-V compressed instructions extension, we need to
+ * check if the two least significant bits of the instruction are
+ * b11 (0x3), which means that the trapped instruction is not compressed,
+ * i.e., the trapped instruction is 32bits = 4bytes. Otherwise, the trapped
+ * instruction is 16bits = 2bytes.
+ */
+
+.balign 4
+.type process_mepc_on_synchronous_irq, @function
+process_mepc_on_synchronous_irq:
```

Comment:
Maybe this deserves a name that is more clear on what it does: `compute_isr_return_address`? Dunno.

Reply:
Changed to `compute_mepc_on_synchronous_irq` since this is just computing the mepc for synchronous IRQs (e.g., exceptions).

Reply:
++